From d1077ff02717174e61ac28c9c9d544c6e68aa8f2 Mon Sep 17 00:00:00 2001
From: Michael-CY Lee <michael-cy.lee@mediatek.com>
Date: Wed, 3 Dec 2025 15:46:08 +0800
Subject: [PATCH 247/254] mtk: hostapd: do not check ACL when handling probe
 request

Because the user might set STA's MLD address in ACL, while the probe
request SA is the STA's link address. The original logic cannot well
handle the situation and IOT issue happens.

Remove the ACL check in the function handle_probe_req().

Signed-off-by: Michael-CY Lee <michael-cy.lee@mediatek.com>
---
 src/ap/beacon.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/ap/beacon.c b/src/ap/beacon.c
index 257cbaa6f..5cf6cdd18 100644
--- a/src/ap/beacon.c
+++ b/src/ap/beacon.c
@@ -1469,7 +1469,9 @@ void handle_probe_req(struct hostapd_data *hapd,
 	int ret;
 	u16 csa_offs[2];
 	size_t csa_offs_len;
+	/* We're not able to retrieve MLD address here
 	struct radius_sta rad_info;
+	*/
 	struct probe_resp_params params;
 	char *hex = NULL;
 #ifdef CONFIG_IEEE80211BE
@@ -1494,6 +1496,7 @@ void handle_probe_req(struct hostapd_data *hapd,
 		sta_track_add(hapd->iface, mgmt->sa, ssi_signal);
 	ie_len = len - IEEE80211_HDRLEN;
 
+	/* We're not able to retrieve MLD address here
 	ret = hostapd_allowed_address(hapd, mgmt->sa, (const u8 *) mgmt, len,
 				      &rad_info, 1);
 	if (ret == HOSTAPD_ACL_REJECT) {
@@ -1502,6 +1505,7 @@ void handle_probe_req(struct hostapd_data *hapd,
 			" due to ACL reject ", MAC2STR(mgmt->sa));
 		return;
 	}
+	*/
 
 	for (i = 0; hapd->probereq_cb && i < hapd->num_probereq_cb; i++)
 		if (hapd->probereq_cb[i].cb(hapd->probereq_cb[i].ctx,
-- 
2.45.2

